scripts/fbi/hate_crime/download_publication_data.py

```diff
@@ -0,0 +1,363 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A script to scrape and download FBI Hate Crime table publications data."""
+import requests
+import os
+from bs4 import BeautifulSoup
+import time
+import json
+from absl import app
+from absl import flags
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer('start_year',
+                     2005,
+                     'Start year(Min. 2005)',
+                     lower_bound=2005)
+flags.DEFINE_integer('end_year', 2019, 'End year(Min. 2005)', lower_bound=2005)
+flags.DEFINE_boolean('force_fetch', True,
+                     'forces api query and not return cached result')
+flags.DEFINE_string('store_path', './tmp', 'Path to store the output')
+flags.DEFINE_string('cache_path', './tmp/.cache_html/',
+                    'Path to store the cached html files')
+flags.DEFINE_string('base_url', 'https://ucr.fbi.gov/hate-crime',
+                    'The base url containing list of URLs for each year')
+
+# TODO(issue id here) store http errors
```

Comment:
Pls open an issue for these.

Reply:
fixed. Created issue https://github.com/datacommonsorg/data/issues/639